tonic/functional/to_averaged_timesurface.py

```diff
@@ -1,146 +1,132 @@
 import numpy as np
-
-
-def findCell(x, y, bounds):
-
-    # build point
-    points = np.array([[x, y]])
-
-    # check for each event if all coordinates are in bounds
-    allInBounds = points[:, 0] >= bounds[:, None, 0]
-    allInBounds &= points[:, 1] >= bounds[:, None, 1]
-    allInBounds &= points[:, 0] < bounds[:, None, 2]
-    allInBounds &= points[:, 1] < bounds[:, None, 3]
-
-    # now find out the positions of all nonzero values
-    nz = np.nonzero(allInBounds)
-
-    # initialize the result with all nan
-    r = np.full(points.shape[0], np.nan)
-
-    # use nz[1] to index event position and nz[0] to tell which cell the event belongs to
-    r[nz[1]] = nz[0]
-    return int(r)
-
-
+from joblib import Parallel, delayed, cpu_count
+import math
+
+def _gen_pixel_to_cell_mapper(w, h, K, px_dtype):
+  matrix = np.zeros((h, w), dtype=px_dtype)
+  for i in range(h):
+    for j in range(w):
+      # Since each frame axis is divided in cells of width _K, the cell index related to that axis can be computed by integer division x//_K.
+      cell_y, cell_x = i//K, j//K
+      # Getting the index associated to cell coordinates.
+      matrix[i,j] = cell_y*math.ceil(w/K) + cell_x
+  return matrix
+  
+def _get_cell_mems(events, n_cells, n_pols, pixel_to_cell_mapper):
+  cell_mems = [[[] for c in range(n_cells)] for p in range(n_pols)]
+  for event in events:
+    cell_mems[max(event['p'], 0)][pixel_to_cell_mapper[event['y'], event['x']]].append(event)
+  return cell_mems[:]
+
+def _bsearch_window(t_start, cell_mem):
+  l, r, start = 0, len(cell_mem)-1, 0
+  while l<=r:
+    m = (l+r)//2
+    if cell_mem[m]['t'] > t_start:
+      r, start = m-1, m
+    elif cell_mem[m]['t'] < t_start:
+      l = m+1
+    else:
+      start = m
+      break
+  return start
```

Comment:
what does this function do?

Reply:
It performes a binary search in the events buffer to find the first event with a timestamp included in the time window. I used it to determine the starting index in the buffer for the time surface computation.